catalogd/internal/storage/index_test.go

```diff
@@ -0,0 +1,285 @@
+							"packageName": "test-package",
+							"version":     "6.0.0",
+						},
+					},
+				},
+			}),
+		},
+	}
+
+	// Create and populate the index
+	metasChan := make(chan *declcfg.Meta, len(metas))
+	for _, meta := range metas {
+		metasChan <- meta
+	}
+	close(metasChan)
+
+	idx := newIndex(metasChan)
+
+	// Create a reader from the metas
+	var combinedBlob bytes.Buffer
+	for _, meta := range metas {
+		combinedBlob.Write(meta.Blob)
+	}
+	fullData := bytes.NewReader(combinedBlob.Bytes())
+
+	tests := []struct {
+		name        string
+		schema      string
+		packageName string
+		blobName    string
+		wantCount   int
+		validate    func(t *testing.T, entry map[string]interface{})
+	}{
+		{
+			name:      "filter by schema - olm.package",
+			schema:    "olm.package",
+			wantCount: 1,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["schema"] != "olm.package" {
+					t.Errorf("Expected olm.package schema blob got %v", entry["schema"])
+				}
+			},
+		},
+		{
+			name:      "filter by schema - olm.channel",
+			schema:    "olm.channel",
+			wantCount: 2,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["schema"] != "olm.channel" {
+					t.Errorf("Expected olm.channel schema blob got %v", entry["schema"])
+				}
+			},
+		},
+		{
+			name:      "filter by schema - olm.bundle",
+			schema:    "olm.bundle",
+			wantCount: 3,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["schema"] != "olm.bundle" {
+					t.Errorf("Expected olm.bundle schema blob got %v", entry["schema"])
+				}
+			},
+		},
+		{
+			name:        "filter by package",
+			packageName: "test-package",
+			wantCount:   5,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["package"] != "test-package" {
+					t.Errorf("Expected blobs with package name test-package, got blob with package name %v", entry["package"])
+				}
+			},
+		},
+		{
+			name:      "filter by specific bundle name",
+			blobName:  "test-bundle.v5.0.3",
+			wantCount: 1,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["schema"] != "olm.bundle" && entry["name"] != "test-bundle.v5.0.3" {
+					t.Errorf("Expected blob with schema=olm.bundle and name=test-bundle.v5.0.3, got %v", entry)
+				}
+			},
+		},
+		{
+			name:        "filter by schema and package",
+			schema:      "olm.bundle",
+			packageName: "test-package",
+			wantCount:   3,
+			validate: func(t *testing.T, entry map[string]interface{}) {
+				if entry["schema"] != "olm.bundle" && entry["package"] != "test-package" {
+					t.Errorf("Expected blob with schema=olm.bundle and package=test-package, got %v", entry)
+				}
+			},
+		},
+		{
+			name:        "no matches",
+			schema:      "non.existent",
+			packageName: "not-found",
+			wantCount:   0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := idx.Get(fullData, tt.schema, tt.packageName, tt.blobName)
```

Comment:
`blobName` and `packageName` are not consistently provided in the tests.  Are they needed?

Reply:
`getSection()` code handles these cases: 

```
func (i *index) getSectionSet(schema, packageName, name string) sets.Set[section] {
	.
        .
	// Filter by package name if specified
	if packageName != "" {
		.
                .
	}

	// Filter by name if specified
	if name != "" {
		.
                .
	}
	return sectionSet
}
```